simpleviewinc/keg-hub
# Spawn-Command with quotes

**Ticket**: [ZEN-416](https://jira.simpleviewtools.com/browse/ZEN-416)

## Context

* This an issue with spawn command handling quoted arguments

## Goal

* Allow spawn command to handle quoted arguments

## Updates

* `repos/spawn-cmd/package.json`
  * Updated jest config to better get coverage results
* `repos/spawn-cmd/src/__tests__/spawnCmd.js`
  * Updated tests for spawn command to test for quoted arguments
* `repos/spawn-cmd/src/utils/__tests__/parseQuotes.js`
  * Added tests for the `parseQuotes` helper
* `repos/spawn-cmd/src/utils/cmdArgs.js`
  * Updated argument parsing to call `parseQuotes` helper
* `repos/spawn-cmd/src/utils/parseQuotes.js`
  * Add helper to search for quoted arguments, and turn them into a single argument


## Testing

* Pull this branch = >`keg &&  keg pr 95`
* Ensure node_modules are installed => `cd repos/spawn-cmd && yarn install`
* Run the tests => `yarn test`
  * Ensure all tests pass
